components/markdown_image_expand/index.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {GlobalState} from 'mattermost-redux/types/store';
+import {connect} from 'react-redux';
+
+import {isEmbedVisible} from 'selectors/posts';
+
+import MarkdownImageExpand, {Props} from './markdown_image_expand';
+
+const mapStateToProps = (state: GlobalState, ownProps: Props) => {
+    return {
+        isEmbedVisible: isEmbedVisible(state, ownProps.postId),
```

Comment:
I found a slight problem with doing this in this way. Since the collpased state is tied to a single post ID, this won't work if you have two Markdown images in a single post since they'll both be looking at the same state for `isEmbedVisible`. Maybe instead of just passing in a `postId`, we also pass in the image URL to differentiate the two images?

The reason this hasn't been a problem before is that we only ever have one embed in the post. Markdown images are different since you can have any number of them in a post.

Reply:
Hmm, actually, that might not work since a user could theoretically post the same Markdown image multiple times in a single post for whatever reason. Perhaps we could do something with the index of the image in the post? I'm not sure if we have any way for the `MarkdownImage` to get that though.

If you could find something like that, that'd be great, but if we're not able to find anything better, we'd probably be fine just sticking with the post ID and image URL together since that would work in most cases.

Reply:
Fixed.